libs/ngx-nuts-and-bolts/src/lib/helpers/table-state/table-state.helper.ts

```diff
@@ -0,0 +1,204 @@
+}
+
+export enum TableQueryParam {
+	PAGE_INDEX = 'pageIndex',
+	PAGE_SIZE = 'pageSize',
+	SORT_KEY = 'sortKey',
+	SORT_DIRECTION = 'sortDirection',
+	CUSTOM_FILTERS = 'filters',
+}
+
+/**
+ *
+ * Creates observable that emits when page information in route changes.
+ * @param {ActivatedRoute} route - Activated route instance used in component.
+ * @param {IPageInfo} [fallbackPaginationInfo] - Fallback values for pagination info.
+ */
+export function createPaginationObservable(
+	route: ActivatedRoute,
+	fallBackPaginationInfo: IPageInfo = { pageIndex: 0, pageSize: 10 }
+): Observable<IPageInfo> {
+	return route.queryParamMap.pipe(
+		map((paramMap) => {
+			const pageIndex = parseInt(
+				paramMap.get(TableQueryParam.PAGE_INDEX) ?? fallBackPaginationInfo.pageIndex.toString(),
+				10
+			);
+			const pageSize = parseInt(
+				paramMap.get(TableQueryParam.PAGE_SIZE) ?? fallBackPaginationInfo.pageSize.toString(),
+				10
+			);
+
+			return {
+				pageIndex,
+				pageSize,
+			};
+		}),
+		distinctUntilChanged(
+			(pageEvent1, pageEvent2) =>
+				pageEvent1.pageIndex === pageEvent2.pageIndex && pageEvent1.pageSize === pageEvent2.pageSize
+		)
+	);
+}
+
+/**
+ *
+ * Creates observable that emits when sorting information in route changes.
+ * @param {ActivatedRoute} route - Activated route instance used in component.
+ */
+export function createSortingObservable(route: ActivatedRoute): Observable<ISortInfo> {
+	return route.queryParamMap.pipe(
+		map((paramMap) => {
+			const sortingKey = paramMap.get(TableQueryParam.SORT_KEY) ?? '';
+			const sortDirection = (paramMap.get(TableQueryParam.SORT_DIRECTION) ?? '') as Sort;
+
+			return {
+				sortingKey,
+				sortDirection,
+			};
+		}),
+		distinctUntilChanged(isEqual)
+	);
+}
+
+/**
+ *
+ * Creates observable that emits when filtering information in route changes.
+ * @param {ActivatedRoute} route - Activated route instance used in component.
+ * @param {Router} router - Router instance used in component.
+ */
+export function createCustomFiltersObservable<TFilterValue extends Record<string, unknown> | null = null>(
+	route: ActivatedRoute,
+	router: Router
+): Observable<TFilterValue | null> {
+	return route.queryParamMap.pipe(
+		map((queryParamMap) => {
+			let filters = null;
+
+			try {
+				if (queryParamMap.has(TableQueryParam.CUSTOM_FILTERS)) {
+					filters = JSON.parse(atob(queryParamMap.get(TableQueryParam.CUSTOM_FILTERS) || ''));
+				}
+			} catch (e) {
+				console.error(e);
+			}
+
+			if (filters && isFilterEmpty(filters)) {
+				clearFilters(router);
+				return null;
+			}
+
+			return filters;
+		}),
+		distinctUntilChanged(isEqual)
+	);
+}
+
+/**
+ *
+ * Creates query params with passed information about pagination.
+ * @param {Router} router - Router instance used in component.
+ * @param {IPageInfo} pageInfo - Object containing information about page size and page index
+ */
+export function onPageChangeHelper(router: Router, pageInfo: IPageInfo): void {
```

Comment:
I would rename these helpers to this:
```
changePage
changeSort
changeFilters
```

Remember to update the example with the new names as well.